cmd/zfs/zfs_main.c

```diff
@@ -5476,7 +5476,6 @@ print_set_creat_perms(uu_avl_t *who_avl)
 		gettext("Create time permissions:\n"),
```

Comment:
It shouldn't be needed: `print_set_creat_perms()` is only called once by `print_fs_perms()` just to print "set" and "creat" permissions (`node->fspn_fsperm.fsp_sc_avl`) which are set in `parse_fs_perm()`:

https://github.com/zfsonlinux/zfs/blob/d1ed1be3cde1be1d4016c9725e9ca0920a3350af/cmd/zfs/zfs_main.c#L4656-L4663

No other type of permissions should ever be present in `who_avl` and i think we can safely assert `weight < 2` here. I can add the following bits just to be on the safe side:

```diff
diff --git a/cmd/zfs/zfs_main.c b/cmd/zfs/zfs_main.c
index 81c49ca..389ab37 100644
--- a/cmd/zfs/zfs_main.c
+++ b/cmd/zfs/zfs_main.c
@@ -5489,6 +5489,7 @@ print_set_creat_perms(uu_avl_t *who_avl)
                deleg_perm_node_t *deleg_node;
 
                if (prev_weight != weight) {
+                       assert(weight < ARRAY_SIZE(sc_title));
                        (void) printf("%s", sc_title[weight]);
                        prev_weight = weight;
                }
```

Reply:
That'd be good, if for no other reason to clearly document it.  We _may_ want to do a little more here simply to make Coverity happy, even if it can't happen.

Reply:
If we were to do a defensive assertion, we would need `assert(weight < ARRAY_SIZE(sc_title) && weight >= 0);`. This is triggering Clang's static analyzer. I am just going to ignore it as a false positive since it is 1 of 12 reports saying `1st/2nd/3rd/nth function call argument is an uninitialized value` and as far as I can tell, every single one is a false positive.